### PR TITLE
feat(core): Implemented support for hideExpression

### PR DESCRIPTION
**Pull request by divyakumarjain**
Implemented support for hideExpression as String expression, boolean and function

#14

### CODE REVIEW

**Comment by divyakumarjain** [comment]
Working sample for hideExpression. Implementation can be done in different way where hideExpression can bind to [hidden]. Tried but it did not work for some reason.

**Comment by mohammedzamakhan** [comment]
Cool, Can you resolve the conflict and push again?

Thanks

File: src/services/formly.field.visibility.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        }
+    }
+    hasHideExpression() :boolean {
+        return this.formlyField.field.hideExpression !== undefined && this.formlyField.field.hideExpression
+    }
+
+    checkVisibilityChange() {
+        var hideExpressionResult: boolean = this.eval(this.formlyField.field.hideExpression);
+        if (this.formlyField.field.hideExpression !== undefined && hideExpressionResult !== this.formlyField.isHidden())  {
+
```
Comment (mohammedzamakhan) [inline src/services/formly.field.visibility.ts]: Remove line

Comment (divyakumarjain) [inline src/services/formly.field.visibility.ts]: I have removed redundant check for hideExpression. But check for change in hideExpressionResult with respect to formlyfield.hide ensures that setHidden is only called when hideExpressionResult has actually changed.

**Pull request merged**